mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,245 @@
+  PARTITION_KEY:         the partition of the target table
+  AUTOMIGRATE:           whether to automigrate staged table
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.commit_table(VARCHAR,VARCHAR,VARCHAR,VARCHAR,VARCHAR,VARCHAR,BOOLEAN);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.commit_table(SOURCE_SCHEMA VARCHAR,
+                                                            SOURCE_TABLE  VARCHAR,
+                                                            TARGET_SCHEMA VARCHAR,
+                                                            TARGET_TABLE  VARCHAR,
+                                                            JOIN_KEY      VARCHAR,
+                                                            PARTITION_KEY VARCHAR,
+                                                            AUTOMIGRATE   BOOLEAN)
+  RETURNS VARCHAR
+  LANGUAGE JAVASCRIPT
+  AS
+  $$
+
+  SOURCE_NAMESPACE = SOURCE_SCHEMA + '.' + SOURCE_TABLE;
+  TARGET_NAMESPACE = TARGET_SCHEMA + '.' + TARGET_TABLE;
+
+  if (AUTOMIGRATE) {
+      /* Create table if doesnt already exists */
+      var create_target_table_query = `CREATE TABLE IF NOT EXISTS ` + TARGET_NAMESPACE + ` AS (SELECT * FROM ` + SOURCE_NAMESPACE + ` WHERE FALSE);`
+      snowflake.createStatement({sqlText: create_target_table_query}).execute();
+      /* Check cols between staging and target and add if required */
+      var cols_check = `CALL {{.output_schema}}.column_check(
+                            '` + SOURCE_SCHEMA + `'
+                          , '` + SOURCE_TABLE + `'
+                          , '` + TARGET_SCHEMA + `'
+                          , '` + TARGET_TABLE + `'
+                          , `  + AUTOMIGRATE.toString() + `);`;
+      snowflake.createStatement({sqlText: cols_check}).execute();
+  }
+
+  /* Prepare delete/insert statements */
+  var trg_columns = list_cols(TARGET_SCHEMA, TARGET_TABLE);
+  var trg_del_condition = `` + JOIN_KEY + ` IN 
+                        (SELECT ` + JOIN_KEY + ` FROM ` + SOURCE_NAMESPACE + `) 
+                        AND ` + PARTITION_KEY + ` >= (SELECT TIMEADD(DAY, -{{or .upsert_lookback_days 30}}, MIN(` + PARTITION_KEY + `)) FROM ` + SOURCE_NAMESPACE + `)`;
+
+  var trg_delete_stmt = `DELETE FROM ` + TARGET_NAMESPACE + ` WHERE ` + trg_del_condition + `;`;
+  var tgr_insert_stmt = `INSERT INTO ` + TARGET_NAMESPACE + ` SELECT ` + trg_columns + ` FROM ` + SOURCE_NAMESPACE + `;`;
+
+  // BEGIN TRANSACTION
+  snowflake.createStatement({sqlText: `BEGIN;`}).execute();
+  try {
+      /* Execute delete and insert */
+      snowflake.createStatement({sqlText: trg_delete_stmt}).execute();
+      snowflake.createStatement({sqlText: tgr_insert_stmt}).execute();
+      snowflake.createStatement({sqlText: `COMMIT;`}).execute();
+
+  } catch(ERROR) {
+
+      snowflake.createStatement({sqlText: `ROLLBACK;`}).execute();
+      throw ERROR;
+
+  }
+  return "ok. commit_staged succeeded.";
+
+  // == Helpers ==
+  function list_cols(sch,tbl) {
```

Comment:
Nit: Not something that has to change this release, but it would be preferable to define our helper functions independently, if possible. An example of why is that currently, we have no way to independently test that `list_cols()` behaves as expected.

Reply:
Will look to strip out the helper functions into their own functions. On this one specifically, I am not convinced we need it at all